src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMSourceContext.java

```diff
@@ -232,7 +253,26 @@ public GitHubSCMSourceContext withForkPRStrategies(Set<ChangeRequestCheckoutStra
         forkPRStrategies.addAll(strategies);
         return this;
     }
+    /**
+     * Defines the strategies used to notify Github of build status.
+     *
+     * @param strategies the strategies used to notify Github of build status.
+     * @return {@code this} for method chaining.
+     * @since TODO
+     */
+    @NonNull
+    public final GitHubSCMSourceContext withNotificationStrategies(List<AbstractGitHubNotificationStrategy> strategies) {
+        notificationStrategies = strategies;
```

Comment:
So we ware using the list as an "ordered set". We should make a defensive copy of the supplied list and enforce that invariant while copying